projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/case-flag-state.service.ts

```diff
@@ -1,10 +1,13 @@
 import { Injectable } from '@angular/core';
 import { FormGroup } from '@angular/forms';
+import { FlagDetailDisplayWithFormGroupPath } from '../../palette/case-flag/domain/case-flag.model';
 
 @Injectable()
 export class CaseFlagStateService {
   public formGroup: FormGroup = new FormGroup({});
   public pageLocation: string;
+  public selectedFlag: FlagDetailDisplayWithFormGroupPath;
```

Comment:
You should be able to access selectedFlag from the formGroup.
Having formGroup and selectedFlag might create issues in the future as we don't have a single source of truth for the selectedFlag.

Reply:
@MateiHMCTS - `CaseFlagFormFields` enum does not have an entry for `selectedFlag`.
Moreover, `selectedFlag` is kind of used as state management to make it available when the user tries to navigate from the Check Your Answers page to previous screens.

Reply:
I understand, but still we're setting it manually in onCaseFlagStateEmitted. It's hooked to the formGroup so you already can access it without setting it. I can see it was previously passed in from the main container.

I would either pass it from the main container or add an entry to the enum and then access it from the FormGroup directly

My concern is that we're setting the state in two places in the stateService which might create issues in the future due to multiple source of truths.

Reply:
@johnbenjamin-hmcts - it does look a bit odd that there's the `selectedFlag` property in the service class but we never see any interactions with it, within the service class itself. Might be worth reconsidering if it can be accessed from the `FormGroup`

Reply:
@MateiHMCTS @Daniel-Lam - I have updated the code, and now using `selectedFlag` from the `formgroup`.